modules/asm/variables.tf

```diff
@@ -40,21 +40,134 @@ variable "gcloud_sdk_version" {
   default     = "296.0.1"
 }
 
-variable "asm_dir" {
-  description = "Name of directory to keep ASM resource config files."
-  type        = string
-  default     = "asm-dir"
-}
+# variable "asm_dir" {
+#   description = "Name of directory to keep ASM resource config files."
+#   type        = string
+#   default     = "asm-dir"
+# }
 
 variable "service_account_key_file" {
   description = "Path to service account key file to auth as for running `gcloud container clusters get-credentials`."
   default     = ""
 }
 
 variable "asm_version" {
-  description = "ASM version to deploy. Available versions are documented in https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages"
+  description = "ASM version to deploy. This module supports versions `1.8` and `1.9`. Available versions are documented in https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages"
+  type        = string
+  default     = "1.9"
+}
+
+variable "asm_git_tag" {
+  description = "ASM git tag to deploy. This module supports versions `1.8` and `1.9`. You can get the exact `asm_git_tag` by running the command `install_asm --version`. The ASM git tab should be of the form `1.9.3-asm.2+config5`. You can also see all ASM git tags by running `curl https://storage.googleapis.com/csm-artifacts/asm/STABLE_VERSIONS`. You must provide the full and exact git tag. This variable is optional. Leaving it empty (default) will download the latest `install_asm` script for the version provided by the `asm_version` variable."
+  type        = string
+  default     = ""
+}
+
+variable "mode" {
+  description = "ASM mode for deployment. Supported mode is `install` only."
   type        = string
-  default     = "1.8"
+  default     = "install"
+}
+
+variable "service_account" {
```

Comment:
Do we need to use a service account to authenticate? I'd prefer if we use whatever credentials Terraform is running as (Application Default Credentials).

Downloading service account keys should not be encouraged.

Reply:
I agree. The install_asm script supports it as an option and this was a feedback from a partner.